test_tools/scripts/record_video_from_camera.py

```diff
@@ -31,15 +32,25 @@ def callback(self, data):
     def clean_shutdown(self):
         if self.video_writer is not None:
             self.video_writer.release()
-        rospy.loginfo(f"Saving video file {self.file_name}")
+            rospy.loginfo(f"Saving video file {self.file_name}")
+        else:
+            rospy.loginfo(f"No video saved, no image received on topic")
 
     def __del__(self):
         self.clean_shutdown()
 
 
 def main():
+    image_topic = "/hero/head_rgbd_sensor/rgb/image_raw"
+    video_name = "video.avi"
+    if len(sys.argv) > 1:
+        image_topic = sys.argv[1]
+        if len(sys.argv) > 2:
+            video_name = sys.argv[2]
```

Comment:
This is not robust for remapping or other ROS args.  Use
```py
myargv = rospy.myargv()
```

P.S. Ideally you use `argparse` for the parsing instead of this positional behaviour.

Reply:
This code will only be run separately. it was never meant to be used with other ros tags

Reply:
No, but someday some idiot will do it.